library/arith.lisp

```diff
@@ -480,6 +480,99 @@ The fields are defined as follows:
       (lisp Double-Float (x y)
         (cl:coerce (cl:/ x y) 'cl:double-float)))))
 
+(coalton-toplevel
+  (define-class (Trigonometric :a)
+    (sin   (:a -> :a))
+    (cos   (:a -> :a))
+    (tan   (:a -> :a))
+    (sinh  (:a -> :a))
+    (cosh  (:a -> :a))
+    (tanh  (:a -> :a))
+    (asin  (:a -> :a))
+    (acos  (:a -> :a))
+    (atan  (:a -> :a))
+    (asinh (:a -> :a))
+    (acosh (:a -> :a))
+    (atanh (:a -> :a)))
+
+  (define-class (Exponentiable :a)
+    (**      (:a -> :a -> :a))
+    (logBase (:a -> :a -> :a)))
+
+  (declare sqrt ((Dividable Integer :a) (Exponentiable :a) => :a -> :a))
+  (define (sqrt x)
+    "Returns the square root of the input"
+    (** x (general/ (the integer 1) (the integer 2))))
+
+  (define-class ((Num :a) (Dividable :a :a) (Exponentiable :a) => Float :a)
+    (e  :a)
+    (pi :a))
+
+  (declare exp ((Float :f) (Exponentiable :f) => :f -> :f))
+  (define (exp x)
+    (** e x))
+
+  (declare log ((Float :f) (Exponentiable :f) => :f -> :f))
+  (define (log x)
+    (logBase e x))
+
+  (define-class ((Float :f) => RealFloat :f)
+    (rationalize (:f -> Fraction))))
+
+(cl:defun %define-unary-wrapper (return-type function cl:&optional (package :cl))
+  "defines a function which is a unary wrapper around a corresponding CL function"
+  (cl:let ((x      (cl:gensym))
```

Comment:
I don't think you need this gensym stuff or the error checking. This is for local use only anyway,

Reply:
I've removed it